### PR TITLE
1665167: Print roles and usage in list of subscriptions; ENT-1315

### PR DESCRIPTION
**Pull request by jirihnidek**
* BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1665167
* List of available packages can contain roles and usage
* TODO: do the same for list of installed packages

### CODE REVIEW

File: test/test_managercli.py [hunk 1]
```diff
@@ -634,6 +637,7 @@ def test_none_wrap_available_pool_id(self, mget_ents):
         def create_pool_list(*args, **kwargs):
             return [{'productName': 'dummy-name',
                      'productId': 'dummy-id',
+                     'roles': 'awsome server',
```
Comment (cnsnyder) [inline test/test_managercli.py]: Non-nacking comment, should we change this from "awsome" to "awesome"?

File: src/rct/printing.py [hunk 2]
```diff
@@ -74,11 +74,14 @@ def as_str(self, order):
         s.append("%s:" % _("Order"))
         s.append("\t%s: %s" % (_("Name"), xstr(order.name)))
         s.append("\t%s: %s" % (_("Number"), xstr(order.number)))
+        s.append("\t%s: %s" % (_("Roles"), xstr(order.roles)))
```
Comment (cnsnyder) [inline src/rct/printing.py]: I think we should not print the header for any attribute which is not included in the source data to ensure compatibility with older versions of Candlepin (both from an entitlement cert content and api perspective).

I think we should list the syspurpose attributes together in the following order after the "Service Type" attribute: Role, Service Level, Usage, Add-ons. As many of the roles (as well as provided products) begin with "Red Hat*" it makes it somewhat difficult to read when the role is listed right after the provided products.

Comment (jirihnidek) [reply src/rct/printing.py]: I agree that attributes not included in entitlement certificate should not be listed in the output of `subscription-manager list --consumed`.

I implemented the order of attributes according proposal from RFE. I thought that John's proposal was based on discussion with you. I like your proposal more. :-)

**Comment by jirihnidek** [comment]
@cnsnyder I improved displaying of syspurpose attributes in `list --consumed`, but I don't like implementation too much, because named tuple is used there. I would like to use dictionary instead, but it would require more refactoring.

**Comment by cnsnyder** [comment]
@jirihnidek If you think we can complete the refactoring by Monday, I'd say go for it, otherwise I'm ok with the current implementation for the moment. Please let me know either way! Thank you!

**Pull request merged**